zzk/service/host.go

```diff
@@ -101,6 +103,8 @@ func (l *HostStateListener) GetPath(nodes ...string) string {
 
 // Ready adds an ephemeral node to the host registry
 func (l *HostStateListener) Ready() error {
+	l.nodelock.Lock()
```

Comment:
The reason I'm locking Ready() is because it can be called from multiple goroutines running the Spawn() function below (that's new with this change). There was no similar change for Done(), so I think we're safe there. As far as #162 goes, I think a lock should be unnecessary, since the Ready() operation is idempotent.